one-ui/src/test/java/com/marklogic/hub/oneui/controllers/SavedQueriesControllerTest.java

```diff
@@ -0,0 +1,215 @@
+                "            \"upperBound\": \"15\"\n" +
+                "          }\n" +
+                "        },\n" +
+                "        \"facet2\": {\n" +
+                "          \"dataType\": \"dateTime\",\n" +
+                "          \"rangeValues\": {\n" +
+                "            \"lowerBound\": \"2020-01-01T13:06:17\",\n" +
+                "            \"upperBound\": \"2020-01-22T13:06:17\"\n" +
+                "          }\n" +
+                "        }\n" +
+                "      }\n" +
+                "    },\n" +
+                "    \"propertiesToDisplay\": [\"facet1\", \"EntityTypeProperty1\"]\n" +
+                "  }\n" +
+                "}";
+        ObjectMapper mapper = new ObjectMapper();
+        queryDoc = mapper.readTree(jsonString);
+    }
+
+    @Test
+    void testSaveNewQuery() {
+        ResponseEntity<JsonNode> response = savedQueriesController.saveQueryDocument(queryDoc);
+        JsonNode responseBody = response.getBody();
+
+        Assertions.assertEquals(201, response.getStatusCodeValue());
+        Assertions.assertNotNull(responseBody);
+        Assertions.assertNotNull(responseBody.get("savedQuery"));
+        Assertions.assertNotNull(responseBody.get("savedQuery").get("id"));
+        Assertions.assertNotNull(responseBody.get("savedQuery").get("systemMetadata"));
+        Assertions.assertEquals("some-query", responseBody.get("savedQuery").get("name").asText());
+        Assertions.assertEquals(4, responseBody.get("savedQuery").get("systemMetadata").size());
+        Assertions.assertEquals("test-data-hub-developer", responseBody.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("test-data-hub-developer", responseBody.get("savedQuery").get("systemMetadata").get("createdBy").asText());
+    }
+
+    @Test
+    void testModifyExistingQuery() {
+        JsonNode firstResponse = savedQueriesController.saveQueryDocument(queryDoc).getBody();
+        ObjectNode savedQueryNode = (ObjectNode) firstResponse.get("savedQuery");
+        String id = savedQueryNode.get("id").asText();
+        savedQueryNode.put("name", "modified-name");
+        ResponseEntity<JsonNode> modifiedResponse = savedQueriesController.updateQueryDocument(firstResponse);
+        JsonNode modifiedResponseBody = modifiedResponse.getBody();
+
+        Assertions.assertEquals(200, modifiedResponse.getStatusCodeValue());
+        Assertions.assertNotNull(modifiedResponseBody);
+        Assertions.assertEquals(id, modifiedResponseBody.get("savedQuery").get("id").asText());
+        Assertions.assertEquals("test-data-hub-developer", modifiedResponseBody.get("savedQuery").get("owner").asText());
+        Assertions.assertEquals("modified-name", modifiedResponseBody.get("savedQuery").get("name").asText());
+    }
+
+    @Test
+    void testCollectionsAndPermissionOnSavedQuery() {
+        JsonNode response = savedQueriesController.saveQueryDocument(queryDoc).getBody();
```

Comment:
There's no need for a separate test here. Just add these assertions to the first test for creating a saved query here.